docs/GHAFileExplained.md

```diff
@@ -0,0 +1,168 @@
+    # build will be created. For example below 6 builds will be created in
+    # total (7.2-pgsql, 7.2-mariadb, 7.3-pgsql, 7.3-mariadb, etc.). If we add
+    # another branch, total number of builds will become 12.
+    strategy:
+      fail-fast: false
+      matrix:
+        php: ['7.2', '7.3', '7.4']
+        moodle-branch: ['MOODLE_310_STABLE']
+        database: [pgsql, mariadb]
+
+    # There is an alterantive way allowing to define explicitly define which php, moodle-branch
+    # and database to use:
+    # matrix:
+    #   include:
+    #     - php: '7.4'
+    #       moodle-branch: 'MOODLE_310_STABLE'
+    #       database: pgsql
+    #     - php: '7.3'
+    #       moodle-branch: 'MOODLE_310_STABLE'
+    #       database: mariadb
+    #     - php: '7.2'
+    #       moodle-branch: 'MOODLE_39_STABLE'
+    #       database: pgsql
+
+    steps:
+      # Check out this repository code in ./plugin directory
+      - name: Check out repository code
+        uses: actions/checkout@v2
+        with:
+          path: plugin
+
+      - name: Install node
+        uses: actions/setup-node@v1
+        with:
+          node-version: '14.15.0'
+
+      # Install PHP of required version. For possible options see https://github.com/shivammathur/setup-php
+      - name: Setup PHP ${{ matrix.php }}
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          coverage: none
+
+      # Install this project into a directory called "ci", updating PATH and
+      # locale.
+      - name: Initialise moodle-plugin-ci
+        run: |
+          composer create-project -n --no-dev --prefer-dist moodlehq/moodle-plugin-ci ci ^3
+          echo $(cd ci/bin; pwd) >> $GITHUB_PATH
+          echo $(cd ci/vendor/bin; pwd) >> $GITHUB_PATH
+          sudo locale-gen en_AU.UTF-8
+
+      # Run the default install.
+      # Optionally, it is possible to specify a different Moodle repo to use
+      # (git://github.com/moodle/moodle.git is used by default) and define
+      # ignore directives or any other env vars for install step.
+      # env:
+      #   MOODLE_REPO=git://github.com/username/moodle.git
+      #   IGNORE_PATHS: 'ignore'
+      #   IGNORE_NAMES: 'ignore_name.php'
+      #   MUSTACHE_IGNORE_NAMES: 'broken.mustache'
+      - name: Install moodle-plugin-ci
+        run: |
+          moodle-plugin-ci install --plugin ./plugin --db-host=127.0.0.1
+        env:
+          DB: ${{ matrix.database }}
+          MOODLE_BRANCH: ${{ matrix.moodle-branch }}
+
+      # Steps that are run for the purpose of testing.  Any of these steps
+      # can be re-ordered or removed to your liking.  And of course, you can
+      # add any of your own custom steps.
+      - name: PHP Lint
+        if: ${{ always() }} # prevents CI run stopping if step failed.
+        run: moodle-plugin-ci phplint
+
+      - name: PHP Copy/Paste Detector
+        continue-on-error: true # This step will show errors but will not fail
```

Comment:
Oh this is nice. So it may be used instead of the current trick based on `|| true` added, right?

Reply:
No.

The error will still fail the overall build. Normally, the build aborts after the first step that fails. This thing tells it to run the other steps anyway (so you get all the falure info.)

Reply:
@mudrd8mz is right, this particular param is needed to override exit code (acts similar to `|| true`). It also prevents build stopping on failure, as name suggest. If all steps have `continue-on-error: true` the build will be successful (green).

Reply:
Thanks.